### ISSUE TITLE
myway.pl fails with 'deep recursion' errors

### ISSUE THREAD

**Issue by srcshelton** [srcshelton/myway#2]
When deploying Vertica schema, myway.pl can fail with:

```
==> xxx: STDERR: Deep recursion on subroutine "myway::checkdbconnection" at /usr/local/bin/myway.pl line 4632.
==> xxx: Deep recursion on subroutine "myway::getsqlvalue" at /usr/local/bin/myway.pl line 4439.
==> xxx: Deep recursion on subroutine "myway::executesql" at /usr/local/bin/myway.pl line 4743.
==> xxx: Deep recursion on subroutine "myway::preparesql" at /usr/local/bin/myway.pl line 4660.
==> xxx: /usr/local/bin/applyschema.sh: line 264: 11255 Killed ${myway} "${params[@]}" "${extraparams[@]}" --init
```

The retry logic on statement execution is:

1. Prepare Statement
1. Execute Statement
1. Confirm database connectivity
1. If connection is dropped then reconnect
1. Re-apply session state (such as SEARCH_PATH)
1. Re-execute failed command => Jump to step 2

... and this error occurs if Vertica persists in dropping the database connection every time statement execution is requested.

We could maintain a counter of the number of times we've retried a statement (which would also indicate the failing statement), but ultimately if the database cannot execute statements via ODBC then we can't deploy to it :(

**Comment by srcshelton** [comment]
We're now iteration-counting and stopping after three attempts. This does nothing to solve that fact that either Vertica, Vertica's ODBC drivers, or Perl's DBI/DBD modules are hosed, but at least the error provided is now a bit friendlier ;)

I've added a further check to try to see whether directly executing a statement has any different a behaviour than the current prepare/execute actions.

**Comment by srcshelton** [comment]
Retry logic fixed in 8685bf9 - Vertica is still problematic :(
